### PR TITLE
[BPK-4292] BPKIcon size APIs

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] [Screenshotting code](https://github.com/Skyscanner/backpack-ios/blob/master/Example/Backpack%20Screenshot/Screenshots.swift)
+ [x] Adding a component? Remember to expose it in the [main `Backpack.h` header file](https://github.com/Skyscanner/backpack-ios/tree/master/Backpack/Backpack.h)
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -488,8 +488,19 @@ gulp.task('generate-icon-names', done => {
       .join('')
       .replace('Ios', 'iOS');
 
-  const templateData = entries =>
-    Object.assign(...entries.map(([k]) => ({ [k]: codify(k) })));
+  // Once we drop support for the legacy API, we can leave the suffix in the string
+  // so that `BPKIcon.m` doesn't need to add it back in programmatically
```
Comment (georgegillams) [inline gulpfile.js:492]: For now while we support both API's I think it's best to cut off the suffix so both API's can use the same underlying mechanism. We can clean up the code in a few weeks when we only need to support the new API.

**Review by k0nserv — COMMENTED** [review]
We had no cases inside Backpack that needed to be updated?

File: Backpack/Icon/Classes/BPKIcon.h [hunk 2]
```diff
@@ -92,5 +97,63 @@ typedef NS_ENUM(NSUInteger, BPKIconSize) {
  * @return The concerete point size for the given size.
  */
 + (CGSize)concreteSizeForIconSize:(BPKIconSize)size NS_SWIFT_NAME(concreteSize(forSize:));
```
[the remark this answers is not in the document]
Comment (georgegillams) [reply Backpack/Icon/Classes/BPKIcon.h:99]: Yeah we will do. Probably just 3 properties:
`concreteSizeForSmallIcon`, `concreteSizeForLargeIcon` and `concreteSizeForXlIcon`?

**Comment by georgegillams** [comment]
> We had no cases inside Backpack that needed to be updated?

Don't want to bloat this PR too much - they can be updated soon 👍

File: Backpack/Icon/Classes/BPKIcon.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (k0nserv) [reply Backpack/Icon/Classes/BPKIcon.h:99]: Yup 👍

**Review by k0nserv — APPROVED** [review]

**Pull request merged**